marklogic-data-hub/src/main/resources/ml-modules/root/marklogic-data-hub-spark-connector/bulkIngester.sjs

```diff
@@ -72,7 +72,9 @@ if (input instanceof Sequence) {
 }
 
 inputArray.forEach(record => {
-  state.next = state.next + 1;
+if(state!=null && state.next!=null){
```

Comment:
@anu3990 I'm approving, but is this even needed anymore?

Reply:
@rjrudin after java-client 5.3 release i removed the code wherein we were creating and sending an empty endpointState. So now, since the endpointState can be null, it was giving NullPointerException at state.next.

Reply:
Can't we remove this entirely though? Our connector doesn't care about anything in the endpointState. And if Ernie wants to store something there, he has to provide his own endpoint.

Reply:
@anu3990 I checked this out locally, and all the tests pass with this block of code removed - i.e. this stuff:

```
if(state!=null && state.next!=null){
    state.next = state.next + 1;
  }
```

As part of rebasing, can you remove that from the endpoint too?